x/ecocredit/basket/msgs.go

```diff
@@ -0,0 +1,51 @@
+package basket
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
+	"github.com/regen-network/regen-ledger/types/math"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+)
+
+var (
+	_ legacytx.LegacyMsg = &MsgPut{}
+
+	_ sdk.Msg = &MsgPut{}
+)
+
+func (m MsgPut) Route() string { return sdk.MsgTypeURL(&m) }
+
+func (m MsgPut) Type() string { return sdk.MsgTypeURL(&m) }
+
+func (m MsgPut) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ecocredit.ModuleCdc.MustMarshalJSON(&m))
+}
+
+func (m MsgPut) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(m.Owner); err != nil {
+		return err
+	}
+	if err := sdk.ValidateDenom(m.BasketDenom); err != nil {
+		return sdkerrors.ErrInvalidRequest.Wrapf("%s is not a valid basket denom", m.BasketDenom)
+	}
+	if len(m.Credits) > 0 {
+		for _, credit := range m.Credits {
+			if err := ecocredit.ValidateDenom(credit.BatchDenom); err != nil {
+				return err
+			}
+			if _, err := math.NewDecFromString(credit.Amount); err != nil {
+				return err
+			}
```

Comment:
How about adding a check that amount is not zero?

Reply:
Must be positive

Reply:
fixed

Reply:
not pushed?

Reply:
was having issues with switching base, should be all good now! thanks!